kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -807,16 +809,44 @@ tryAxiomClaimWorker mode ref = do
             getAxiomOrClaimByName
             ref
     case maybeAxiomOrClaim of
-       Nothing ->
-           putStrLn' "Could not find axiom or claim."
-       Just axiomOrClaim -> do
-           node <- Lens.use (field @"node")
-           case mode of
-               Never ->
-                   showUnificationFailure axiomOrClaim node
-               IfPossible ->
-                   tryForceAxiomOrClaim axiomOrClaim node
+        Nothing ->
+            putStrLn' "Could not find axiom or claim."
+        Just axiomOrClaim -> do
+            claim <- Lens.use (field @"claim")
+            if isReachabilityRule claim && not (equalClaimTypes axiomOrClaim claim)
```

Comment:
This could be `either (const True) (equalClaimTypes claim)`, and that would make `equalClaimTypes :: claim -> claim -> Bool` which makes more sense to me.

PS: `equalClaimTypes` could be implemented as
```
isSameType `on` castToReachability
```